python/cudf_polars/cudf_polars/dsl/utils/windows.py

```diff
@@ -0,0 +1,197 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Utilities for rolling window aggregations."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pyarrow as pa
+
+import polars as pl
+
+import pylibcudf as plc
+
+if TYPE_CHECKING:
+    from cudf_polars.typing import ClosedInterval, Duration
+
+
+__all__ = [
+    "duration_to_int",
+    "duration_to_scalar",
+    "offsets_to_windows",
+    "range_window_bounds",
+]
+
+
+def duration_to_int(
+    dtype: plc.DataType,
+    months: int,
+    weeks: int,
+    days: int,
+    nanoseconds: int,
+    parsed_int: bool,  # noqa: FBT001
+    negative: bool,  # noqa: FBT001
+) -> int:
+    """
+    Convert a polars duration value to an integer.
+
+    Parameters
+    ----------
+    dtype
+        The type of the column being added to.
+    months
+        Number of months
+    weeks
+        Number of weeks
+    days
+        Number of days
+    nanoseconds
+        Number of nanoseconds
+    parsed_int
+        Is this actually a representation of an integer, not a duration?
+    negative
+        Is this a negative duration?
+
+    Returns
+    -------
+    Pyarrow scalar
```

Comment:
Does this return a pyarrow scalar, or a python int? Or maybe the scalar we return subclasses int?

Do we have to worry about overflows here (or in the functions using this)?

Reply:
A python int. But it's an int that comes from within polars, so we don't have to worry about overflow.